Dockerfile

```diff
@@ -24,8 +24,8 @@ ENV COMMIT_SHA=${COMMIT_SHA:-}
 ENV REACT_APP_BUILD_COMMIT_SHA=${COMMIT_SHA:-}
 
 # Download dependencies
-COPY ui /app
```

Comment:
I want to have the same file structure inside and outside the container. It's easy to find a file, like scripts. I agree that it increases the layer size. I will optimize for the final running image. But for building and development, I prefer simple scripts and easy development.

Reply:
Even if you'll use stage-1 images to run the services for front and back, what would be the procedure of applying the changes, made by the developer in the source code?

Right now it looks to me that after each change, which the developer wants to see on the product, the corresponding image should be rebuilt (compose stop service, compose build service, compose start service) - this has not so many benefits compared to building the PROD image locally.

On the other hand, if we would use (or build on top of it) air and nodeamon images to dynamically build the source code from the corresponding volumes, there is no need in coping the source code to the image at the build time, as it will be mounted as a volume and available at container's run time.

Reply:
In `docker-compose.yaml`, it mounts the current directory to `/app` in the container, which overwrites what copies here. Developer could run `docker compose run api /bin/bash` and `go build .` to build the current code.

I plan to add some fs monitor tools later, but I don't know which commands to run when files change. For me, I hope to run tests, but I guess others may want to rebuild and run the server.

Reply:
This topic was discussed in a few issues on the repo, and the main request was to have the dynamically rebuilt server, watching for source code changes. I agree that the environment to safely run tests is important too, but I don't remember these requests.

As there is the volume with the source code to the service, why do we need to copy the repo inside the image at build time?

Reply:
Split into another stage `env` for developing.

Reply:
I'll work on fs monitor in the next step.